__unported__/base_delivery_carrier_files/carrier_file.py

```diff
@@ -76,17 +57,17 @@ def _write_file(self, cr, uid, carrier_file, filename, file_content,
         :return: True if write is successful
         """
         if not carrier_file.export_path:
-            raise orm.except_orm(_('Error'),
-                                 _('Export path is not defined '
-                                   'for carrier file %s') %
-                                 (carrier_file.name,))
+            raise exceptions.except_orm(_('Error'),
+                                        _('Export path is not defined '
+                                          'for carrier file %s') %
+                                        (carrier_file.name,))
         full_path = os.path.join(carrier_file.export_path, filename)
         with open(full_path, 'w') as file_handle:
             file_handle.write(file_content)
         return True
 
-    def _generate_files(self, cr, uid, carrier_file, picking_ids,
-                        context=None):
+    @api.multi
+    def _generate_files(self, carrier_file, picking_ids):
```

Comment:
If this is a api.multi, you should rewrite it to use `self` instead of `carrier_file`